pkg/trace/skywalking/http/tracer.go

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package http
+
+import (
+	"context"
+	"github.com/SkyAPM/go2sky"
+	"github.com/SkyAPM/go2sky/propagation"
+	"github.com/SkyAPM/go2sky/reporter/grpc/common"
+	"mosn.io/api"
+	"mosn.io/mosn/pkg/log"
+	"mosn.io/mosn/pkg/protocol"
+	"mosn.io/mosn/pkg/protocol/http"
+	"mosn.io/mosn/pkg/trace"
+	"mosn.io/mosn/pkg/trace/skywalking"
+	"mosn.io/mosn/pkg/types"
+	"strconv"
+	"time"
+)
+
+var (
+	// MIME header key s. The canonicalization converts the first letter and any letter following a hyphen to upper case;
+	sw6Header = [2]string{propagation.Header, "Sw6"}
+)
+
+func init() {
+	trace.RegisterTracerBuilder("SkyWalking", protocol.HTTP1, NewHttpSkyTracer)
+}
+
+type httpSkyTracer struct {
+	*go2sky.Tracer
+}
+
+func NewHttpSkyTracer(_ map[string]interface{}) (types.Tracer, error) {
+	return &httpSkyTracer{}, nil
+}
+
+func (tracer *httpSkyTracer) SetGO2SkyTracer(t *go2sky.Tracer) {
+	tracer.Tracer = t
+}
+
+func (tracer *httpSkyTracer) Start(ctx context.Context, request interface{}, _ time.Time) types.Span {
+	header, ok := request.(http.RequestHeader)
+	if !ok || header.RequestHeader == nil {
+		log.DefaultLogger.Debugf("[SkyWalking] [tracer] [http1] unable to get request header, downstream trace ignored")
+		return skywalking.NoopSpan
+	}
+
+	// create entry span (downstream)
+	requestURI := string(header.RequestURI())
+	entry, nCtx, err := tracer.CreateEntrySpan(ctx, requestURI, func() (sw6 string, err error) {
+		for _, h := range sw6Header {
+			sw6, ok = header.Get(h)
+			if ok {
+				// delete the sw6 header, otherwise the upstream service will receive two sw6 header
+				header.Del(h)
+				return sw6, err
+			}
+		}
+		return
+	})
+	if err != nil {
+		log.DefaultLogger.Errorf("[SkyWalking] [tracer] [http1] create entry span error, err: %v", err)
+		return skywalking.NoopSpan
+	}
+	entry.Tag(go2sky.TagHTTPMethod, string(header.Method()))
+	entry.Tag(go2sky.TagURL, string(header.Host())+requestURI)
```

Comment:
Host maybe is nil that HTTP1.0， it‘s ok？

Reply:
I've changed the code.